File: Java/프로그래머스/2/160585. 혼자서 하는 틱택토/혼자서 하는 틱택토.java
class Solution {
    public int solution(String[] board) {
        int answer = 1;
        int oCount = 0;
        int xCount = 0;
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[0].length(); j++) {
                if (board[i].charAt(j) == 'O') {
                    oCount++;
                } else if (board[i].charAt(j) == 'X') {
                    xCount++;
                }
            }
        }
        if (xCount > oCount) {
            return 0;
        }
        if (oCount > xCount + 1) {
            return 0;
        }
        if (isComplete(board, 'O')) {
            if (oCount == xCount) {
                return 0;
            }
        }
        if (isComplete(board, 'X')) {
            if (oCount == xCount + 1) {
                return 0;
            }
        }
        return answer;
    }

    boolean isComplete(String[] board, char c) {
        for (int i = 0; i < board.length; i++) {
            if (board[i].charAt(0) == c && board[i].charAt(1) == c && board[i].charAt(2) == c) {
                return true;
            }
        }

        for (int i = 0; i < board[0].length(); i++) {
            if (board[0].charAt(i) == c && board[1].charAt(i) == c && board[2].charAt(i) == c) {
                return true;
            }
        }

        if (board[0].charAt(0) == c && board[1].charAt(1) == c && board[2].charAt(2) == c) {
            return true;
        }
        if (board[0].charAt(2) == c && board[1].charAt(1) == c && board[2].charAt(0) == c) {
            return true;
        }

        return false;
    }
}

File: Python/프로그래머스/2/160585. 혼자서 하는 틱택토/혼자서 하는 틱택토.py
def solution(board):
    def is_bingo(c):
        nonlocal board
        for i in range(len(board)):
            r_cnt, c_cnt = 0, 0
            for j in range(len(board[0])):
                if board[i][j] == c:
                    r_cnt += 1
                if board[j][i] == c:
                    c_cnt += 1
            if r_cnt == len(board) or c_cnt == len(board):
                return True

        d_cnt, dd_cnt = 0, 0
        for i in range(len(board)):
            if board[i][i] == c:
                d_cnt += 1
            if board[i][len(board) - 1 - i] == c:
                dd_cnt += 1
        if d_cnt == len(board) or dd_cnt == len(board):
            return True

        return False

    o_cnt, x_cnt = 0, 0
    for row in board:
        for c in row:
            if c == "O":
                o_cnt += 1
            elif c == "X":
                x_cnt += 1

    answer = 0
    if o_cnt == x_cnt:
        if not is_bingo("O"):
            answer = 1
    elif o_cnt == x_cnt + 1:
        if not is_bingo("X"):
            answer = 1

    return answer